src/sprites/setup.py

```diff
@@ -0,0 +1,230 @@
+    def __init__(self, default: pygame.Rect):
+        """
+        This class can be used to specify custom Sprite hitboxes.
+
+        Use this classes methods to specify exceptions (conditions under which
+        a hitbox different to the default one should be used)
+
+        :param default: The default hitbox to use, when no hitbox has been
+                        found for a specific state / direction
+        """
+
+        self.default = default
+
+        self._state_exceptions = {}
+        self._direction_exceptions = {}
+        self._exceptions = {}
+
+    def set_direction_exception(
+            self, direction: Direction, rect: pygame.Rect
+    ) -> None:
+        """
+        Tell the class, in which direction a hitbox should be used that differs
+        from the default one.
+
+        Hitboxes that were stored through this method will be used before any
+        hitbox specific to the current Entity state is considered.
+        """
+        self._direction_exceptions[direction] = rect
+
+    def set_state_exception(
+            self, state: EntityState, rect: pygame.Rect
+    ) -> None:
+        """
+        Tell the class, in which state a hitbox should be used that differs
+        from the default one.
+        """
+        self._state_exceptions[state] = rect
+
+    def set_exception(
+            self, state: EntityState, direction: Direction, rect: pygame.Rect
+    ) -> None:
+        """
+        Tell the class, in which state a hitbox should be used that differs
+        from the default one.
+
+        Hitboxes that were stored through this method will be used before any
+        hitbox specific to the current Entity state or direction is considered.
+        """
+        self._exceptions[(state, direction)] = rect
+
+    def get_hitbox(
+            self, state: EntityState, direction: Direction
+    ) -> pygame.Rect:
+        state_exception = self._state_exceptions.get(state)
+        direction_exception = self._direction_exceptions.get(direction)
+        exception = self._exceptions.get((state, direction))
+        if not exception:
+            if not direction_exception:
+                if not state_exception:
+                    return self.default
+                return state_exception
+            return direction_exception
+        return exception
+
+    def scale_hitboxes(self, factor: float) -> None:
+        print(self._state_exceptions, self._direction_exceptions, self._exceptions, self.default)
+        for item in [self._state_exceptions.values(),
+                     self._direction_exceptions.values(),
+                     self._exceptions.values(),
+                     [self.default]]:
+            for hitbox in item:
+                hitbox.update(
+                    hitbox.left * factor,
+                    hitbox.top * factor,
+                    hitbox.width * factor,
+                    hitbox.height * factor
+                )
+
+
+class EntityAssets:
+    Rabbit: EntityAsset
```

Comment:
In this case, I assume it's unlikely that EntityAssets.Rabbit is going to get changed after its creation. As such, it should be better, coding convention-wise, to use an all-caps name (EntityAssets.RABBIT) instead of the current one.

```suggestion
    RABBIT: EntityAsset
```

Reply:
Alright, sounds reasonable 😅